pilot/pkg/model/jwks_resolver.go

```diff
@@ -242,16 +245,12 @@ func (r *JwksResolver) GetPublicKey(issuer string, jwksURI string) (string, erro
 	if jwksURI == "" {
 		// Fetch the jwks URI if it is not hardcoded on config.
 		jwksURI, err = r.resolveJwksURIUsingOpenID(issuer)
+		key.jwksURI = jwksURI
```

Comment:
using the return value (jwksURI) when an `err` was returned is not a great practice IMO. If you are intentionally setting it to `""` on err, then I would make it explicit.